server/src/main/java/io/crate/planner/node/ddl/CreateTablePlan.java

```diff
@@ -231,4 +243,15 @@ private static boolean hasColumnDefinition(AnalyzedTableElements tableElements,
         return (tableElements.columnIdents().contains(columnIdent) ||
                 columnIdent.name().equalsIgnoreCase("_id"));
     }
+
+    private static void validateSoftDeletesSetting(TableParameter tableParameter, ClusterState clusterState) {
+        var softDeleteEnabled = IndexSettings.INDEX_SOFT_DELETES_SETTING.get(tableParameter.settings());
+        if (softDeleteEnabled == false) {
+            var version = clusterState.getNodes().getLocalNode().getVersion();
+            if (version.onOrAfter(Version.V_5_0_0)) {
+                throw new IllegalArgumentException("Soft deletes must be enabled in version " + version);
+            }
```

Comment:
Could this validation get moved into the setting definition? (Settings support validators)
Would have the advantage that rules regarding the setting are in a single place.

I am also not sure if this handles the mixed cluster case correctly. I think the elected master sets the "final" index version - so if there is a mixed cluster with master node being on 5.0 and a handler node being 4.x, the validation here wouldn't trigger, but the effective index-version would be 5.0. 

Sorry - should've been more verbose about this, but this is what I meant with it depending on the index version.

Reply:
The version handling here is wrong. The index version on the master node is determined by the oldest (lowest) version of nodes in the cluster. This was supposed to be a draft anyway, i marked it by mistake for review.  I will do one more iteration on this.